docs/planning/backend/app/api/imports.py

```
@@ -0,0 +1,144 @@
+    """
+    if not file.filename.endswith('.csv'):
+        raise HTTPException(status_code=400, detail="File must be a CSV")
+
+    # Save uploaded file temporarily
+    data_dir = Path(settings.DATA_DIR)
+    data_dir.mkdir(parents=True, exist_ok=True)
+
+    temp_path = data_dir / f"upload_{file.filename}"
+
+    try:
+        with open(temp_path, 'wb') as buffer:
```

Ruff (0.14.10)

Async functions should not open files with blocking methods like `open`

(ASYNC230)